[`pylint`] make fix unsafe if delete comments (`PLR1730`)

VascoSch92 opened the pull request:
The PR dresses issue #17311

ntBre commented on crates/ruff_linter/src/rules/pylint/rules/if_stmt_min_max.rs:
````diff
@@ -29,6 +29,19 @@ use crate::fix::snippet::SourceCodeSnippet;
+/// ## Fix safety
+/// This fix is marked unsafe if it would delete any comments within the replacement range.
+///
+/// An example to illustrate where comments are preserved and where they are not:
+///
+/// ```pyi
+/// a, b = 0, 10
+///
+/// if a >= b: # deleted comment
+///     # deleted comment
+///     a = b # preserved comment
+/// ```
+///
````
I don't think we generally include examples like this, but I don't really mind it if you want to keep it. The `preserved comment` here is technically after the replacement range, so it's a bit redundant with the explanation in your first line, but that's probably not obvious to most users.
```suggestion
/// An example to illustrate where comments are preserved and where they are not:
///
/// ```py
/// a, b = 0, 10
///
/// if a >= b: # deleted comment
///     # deleted comment
///     a = b # preserved comment
/// ```
///
```

If we do keep the example, I think this should just be `py`. I believe `pyi` is for stub files, where code like this would be pretty unusual, as far as I know.

ntBre commented on crates/ruff_linter/src/rules/pylint/rules/if_stmt_min_max.rs:
```diff
@@ -169,11 +182,17 @@ pub(crate) fn if_stmt_min_max(checker: &Checker, stmt_if: &ast::StmtIf) {
         stmt_if.range(),
     );
 
+    let applicability = if checker.comment_ranges().intersects(stmt_if.range()) {
```
nit: we should factor out `stmt_if.range()` into a variable since we use it here and in the `Edit::range_replacement` call.

ntBre reviewed (approved):
Thanks! This looks good, just two nits (and you may need to reformat the example code block to pass CI, if we decide to keep it)

VascoSch92 commented:
Hey @ntBre 
I addressed your feedbacks. I hope it is what you wanted. 

For the example: I took inspiration from [here](https://github.com/VascoSch92/ruff/blob/44ad2012622d73373eb7922978f391eb6c1e54fa/crates/ruff_linter/src/rules/flake8_pyi/rules/unnecessary_literal_union.rs#L36) 
But I can delete it. No problem ;-)

ntBre commented:
> Hey @ntBre I addressed your feedbacks. I hope it is what you wanted.

Thanks!

> For the example: I took inspiration from [here](https://github.com/VascoSch92/ruff/blob/44ad2012622d73373eb7922978f391eb6c1e54fa/crates/ruff_linter/src/rules/flake8_pyi/rules/unnecessary_literal_union.rs#L36) But I can delete it. No problem ;-)

That rule applies to stub files, and  the `## Example` section also uses `pyi`. This rule is for non-stub files, so we should use `py` or `python` in the code blocks.

> PS: the CI is failing because of some red knot stuff. I don't know how this is related to my changes

Yeah no worries, `main` was actually failing for a little while, and your PR must have picked that up. I'll close and re-open the PR to retrigger CI because I think this is good to land otherwise.
